tempesta_fw/http_frame.c

```diff
@@ -0,0 +1,1017 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2014 NatSys Lab. (info@natsys-lab.com).
+ * Copyright (C) 2015-2019 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#if DBG_HTTP_FRAME == 0
+#undef DEBUG
+#endif
+#include "lib/fsm.h"
+#include "lib/str.h"
+#include "procfs.h"
+#include "http.h"
+#include "http_frame.h"
+
+
+#define FRAME_CLI_MAGIC		"PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n"
+#define FRAME_CLI_MAGIC_LEN	24
+#define FRAME_SRVC1_SIZE	4
+#define FRAME_PRIORITY_SIZE	5
+#define FRAME_STNGS_ENTRY_SIZE	6
+#define FRAME_SRVC2_SIZE	8
+
+#define FRAME_STREAM_ID_MASK	((1U << 31) - 1)
+
+/* HTTP/2 frame types (RFC 7540 section 6).*/
+typedef enum {
+	HTTP2_DATA		= 0,
+	HTTP2_HEADERS,
+	HTTP2_PRIORITY,
+	HTTP2_RST_STREAM,
+	HTTP2_SETTINGS,
+	HTTP2_PUSH_PROMISE,
+	HTTP2_PING,
+	HTTP2_GOAWAY,
+	HTTP2_WINDOW_UPDATE,
+	HTTP2_CONTINUATION
+} TfwFrameType;
+
+/*
+ * HTTP/2 error codes (RFC 7540 section 7). Used in RST_STREAM
+ * and GOAWAY frames to report the reasons of the stream or
+ * connection error.
+ */
+#define FRAME_ECODE_PROTO	0x1
```

Comment:
Section 7 contains 14 error codes. I understand that only two are needed at this moment, but it seems that we will definitely will need all of them, let's introduce enum like `TfwFrameType` to store all the possible errors.

Reply:
Done.